test/unittest/xmlparser/XMLProfileParserTests.cpp

```diff
@@ -1625,6 +1626,438 @@ TEST_F(XMLProfileParserTests, external_locators_feature)
+                    </metatraffic_external_unicast_locators>
+                </builtin>
+            </rtps></participant></profiles>)",
+            xmlparser::XMLP_ret::XML_ERROR
+        },
+        {
+            "participant_complete_wrong_loc_udp6_multicast",
+            R"(<profiles><participant profile_name="p1"><rtps>
+                <ignore_non_matching_locators>false</ignore_non_matching_locators>
+                <default_external_unicast_locators>
+                    <udpv4 externality="1" cost="0" mask="24">
+                        <address>192.168.1.100</address>
+                        <port>1234</port>
+                    </udpv4>
+                    <udpv6 externality="1" cost="0" mask="24">
+                        <address>ff1e::ffff:efff:1</address>
+                        <port>1235</port>
+                    </udpv6>
+                </default_external_unicast_locators>
+                <builtin>
+                    <metatraffic_external_unicast_locators>
+                        <udpv4 externality="1" cost="0" mask="24">
+                            <address>192.168.1.100</address>
+                            <port>2234</port>
+                        </udpv4>
+                        <udpv6 externality="1" cost="0" mask="24">
+                            <address>::1:2</address>
+                            <port>2235</port>
+                        </udpv6>
+                    </metatraffic_external_unicast_locators>
+                </builtin>
+            </rtps></participant></profiles>)",
+            xmlparser::XMLP_ret::XML_ERROR
+        },
+        {
+            "participant_complete_wrong_loc_empty",
+            R"(<profiles><participant profile_name="p1"><rtps>
+                <ignore_non_matching_locators>false</ignore_non_matching_locators>
+                <default_external_unicast_locators>
+                    <udpv4 externality="1" cost="0" mask="24">
+                    </udpv4>
+                    <udpv6 externality="1" cost="0" mask="24">
+                        <address>::1:2</address>
+                        <port>1235</port>
+                    </udpv6>
+                </default_external_unicast_locators>
+                <builtin>
+                    <metatraffic_external_unicast_locators>
+                        <udpv4 externality="1" cost="0" mask="24">
+                            <address>192.168.1.100</address>
+                            <port>2234</port>
+                        </udpv4>
+                        <udpv6 externality="1" cost="0" mask="24">
+                            <address>::1:2</address>
+                            <port>2235</port>
+                        </udpv6>
+                    </metatraffic_external_unicast_locators>
+                </builtin>
+            </rtps></participant></profiles>)",
+            xmlparser::XMLP_ret::XML_ERROR
+        },
+        {
+            "participant_complete_wrong_loc_no_addr",
```

Comment:
Similar to this test. What would happen if I set two \<address\> or \<port\> tags accidentally in my locator? Should it fail?

Reply:
It should. I will add a test for it.

Reply:
Done in fe7ce528fc56b0c0f9a39be01044409711d7abf9